Microsoft.Toolkit.Uwp.Services/Services/MicrosoftTranslator/AzureAuthToken.cs

```diff
@@ -0,0 +1,133 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Microsoft.Toolkit.Uwp.Services.MicrosoftTranslator
+{
+    /// <summary>
+    /// Client to call Cognitive Services Azure Auth Token service in order to get an access token.
+    /// </summary>
+    internal class AzureAuthToken
+    {
+        /// <summary>
+        /// Name of header used to pass the subscription key to the token service
+        /// </summary>
+        private const string OcpApimSubscriptionKeyHeader = "Ocp-Apim-Subscription-Key";
+
+        /// <summary>
+        /// URL of the token service
+        /// </summary>
+        private static readonly Uri ServiceUrl = new Uri("https://api.cognitive.microsoft.com/sts/v1.0/issueToken");
+
+        /// <summary>
+        /// After obtaining a valid token, this class will cache it for this duration.
+        /// Use a duration of 5 minutes, which is less than the actual token lifetime of 10 minutes.
+        /// </summary>
+        private static readonly TimeSpan TokenCacheDuration = new TimeSpan(0, 5, 0);
```

Comment:
Because this value is used to control when to refresh authentication token (as reported in the comments). Setting it to an incorrect value can prevent the service to work properly.